src/Utils/Hooks/useClientQuery.ts

```diff
@@ -50,6 +52,14 @@ export const useClientQuery = <T extends OperationType>({
 
       setData(res)
       setLoading(false)
+
+      const operation = createOperationDescriptor(
+        getRequest(query),
+        variables ?? {}
+      )
+
+      // Retain the operation to prevent it from being garbage collected. Garbage collection can compromise type safety (e.g. non-nullable values being `null`), potentially leading to runtime errors.
+      relayEnvironment.retain(operation)
```

Comment:
lets return the disposable instance in the return value of the hook:

```tsx
const disposable = relayEnvironment.retain(operation)
...
return {
  ... 
  dispose: disposable.dispose 
}
```

Reply:
Good idea 👍